sycl/include/sycl/ext/oneapi/weak_object_base.hpp

```diff
@@ -0,0 +1,66 @@
+//==------- weak_object_base.hpp --- SYCL weak objects base class ----------==//
```

Comment:
nit: I'd prefer not having this separate file, but up to you.

Reply:
It is needed because we specialize `weak_object` for buffer and that would either require us to create a source file or create a circular dependency, so this seemed like the simpler path.